### PR TITLE
[makeotfexe] confirm that nextNode is not NULL before using it

### LINKED ISSUE #993: [makeotfexe] Crash with contextual pos rule
The following (which I’m not sure is valid) crashes makeotfexe:
```fea
markClass [acute grave] <anchor 150 -10> @TOP_MARKS;                            
                                                                                
lookup mark {                                                                   
  position base [a e o u]' <anchor 250 450> mark @TOP_MARKS;                    
} mark;
```

### PR DESCRIPTION
**Pull request by cjchapman**
fixes #993

### CODE REVIEW

File: tests/makeotfexe_data/expected_output/bug993.ttx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+          <!-- BacktrackGlyphCount=0 -->
+          <!-- InputGlyphCount=1 -->
+          <InputCoverage index="0" Format="1">
+            <Glyph value="a"/>
+            <Glyph value="e"/>
+            <Glyph value="o"/>
+            <Glyph value="u"/>
+          </InputCoverage>
+          <!-- LookAheadGlyphCount=0 -->
+          <!-- PosCount=0 -->
```
Comment (anthrotype) [inline tests/makeotfexe_data/expected_output/bug993.ttx]: this ChainingContextPos lookup is not doing anything. Its array of PosLookupRecords is empty.

Comment (khaledhosny) [reply tests/makeotfexe_data/expected_output/bug993.ttx]: What @anthrotype said. This does not seem to be doing anything.

Comment (cjchapman) [reply tests/makeotfexe_data/expected_output/bug993.ttx]: OK, that seems to be by design, as this produces it:
```
  position base [a e o u]' <anchor 250 450> mark @TOP_MARKS;                    
```
...but this (without the apostrophe) does not:
```
  position base [a e o u] <anchor 250 450> mark @TOP_MARKS;                    
```
...so I've asked @readroberts for his guidance on this.

**Comment by cjchapman** [comment]
@readroberts OK. I changed both checks to throw a fatal error on NULL pointers.

**Pull request merged**